src/ui/public/agg_types/metrics/cumulative_sum.js

```diff
@@ -0,0 +1,17 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import ParentPipelineAggHelperProvider from './lib/parent_pipeline_agg_helper';
+import { makeNestedLabel } from './lib/make_nested_label';
+
+export default function AggTypeMetricComulativeSumProvider(Private) {
+  const MetricAggType = Private(AggTypesMetricsMetricAggTypeProvider);
+  const parentPipelineAggHelper = Private(ParentPipelineAggHelperProvider);
+
+  return new MetricAggType({
+    name: 'cumulative_sum',
+    title: 'Cumulative Sum',
+    makeLabel: agg => makeNestedLabel(agg, 'cumulative sum'),
+    params: [
+      ...parentPipelineAggHelper.params()
```

Comment:
why i don't want to refactor this further: at this point they are indeed the same (cumulative sum and derivative) but they actually have different sub options (which we don't expose right now but probably will in the future).

with the current form it'll be really easy to add the parameters that are different + still not be duplicating logic for the parameters that are the same.